src/unstract/adapters/vectordb/qdrant/src/qdrant.py

```diff
@@ -58,23 +56,17 @@ def get_vector_db_instance(self) -> Optional[BasePydanticVectorStore]:
                 self.config.get(VectorDbConstants.EMBEDDING_DIMENSION),
             )
             url = self.config.get(Constants.URL)
-            if (
-                self.config.get(Constants.API_KEY) is not None
-                or self.config.get(Constants.API_KEY) == ""
-            ):
-                self.client = QdrantClient(
-                    url=url, api_key=self.config.get(Constants.API_KEY)
-                )
-                vector_db = QdrantVectorStore(
-                    collection_name=self.collection_name,
-                    client=self.client,
-                    api_key=self.config.get(Constants.API_KEY),
-                )
-            else:
-                self.client = QdrantClient(url=url)
-                vector_db = QdrantVectorStore(
-                    collection_name=self.collection_name, client=self.client
-                )
+            user_api_key: Optional[str] = self.config.get(
+                Constants.API_KEY, None
+            )
+            api_key: Optional[str] = user_api_key if user_api_key else None
+            if api_key:
```

Comment:
NIT: Good to throw an error if this is not provided. I don't think we do JSON schema validation for these adapters forms

Reply:
user_api_key is just a placholder for self.config.get(Constants.API_KEY, None). I did not want line 62 to this snippet twise. So just used a variable. No other logic.

Reply:
For locally hosted qdrant, at one point i was able to configure without providing any api key. hence this was initially kept optional.  Not sure why this changed at some point. Need to test and see if we are still able to connect to local qdrant without api key.

Reply:
It'll be good if we can check it and make changes necessary for that. Any additional / dead code would only lead to confusion later

Reply:
Done. The code now creates the client with api key only if user provides any value.